### ISSUE TITLE
Why prow jobs which setup multicluster have `TEST_SELECT="-multicluster"`?

### ISSUE THREAD

**Issue by su225** [istio/istio#34517]
**Bug description**
As far as I understand, `TEST_SELECT` with something like `-multicluster` actually excludes multicluster tests. Here are some examples from the test-infra repo

Source:
https://github.com/istio/istio/blob/master/tests/integration/README.md#test-selection

https://github.com/istio/test-infra/blob/e98f7864b19888a1246612391a06c1a46f9a527f/prow/config/jobs/istio.yaml#L65-L76
https://github.com/istio/test-infra/blob/e98f7864b19888a1246612391a06c1a46f9a527f/prow/config/jobs/istio.yaml#L78-L95
https://github.com/istio/test-infra/blob/e98f7864b19888a1246612391a06c1a46f9a527f/prow/config/jobs/istio.yaml#L170-L181
https://github.com/istio/test-infra/blob/e98f7864b19888a1246612391a06c1a46f9a527f/prow/config/jobs/istio.yaml#L183-L200
https://github.com/istio/test-infra/blob/e98f7864b19888a1246612391a06c1a46f9a527f/prow/config/jobs/istio.yaml#L216-L227
https://github.com/istio/test-infra/blob/e98f7864b19888a1246612391a06c1a46f9a527f/prow/config/jobs/istio.yaml#L229-L246

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[X] Developer Infrastructure
[ ] Upgrade

**Comment by howardjohn** [comment]
cc @stevenctl

**Comment by stevenctl** [comment]
This really isn't needed anymore. 

We used to set a label on suites like this:

https://github.com/istio/istio/blob/501775a7170c4823e5e45c244aeeb8388753034d/tests/integration/multicluster/base/main_test.go#L36

But instead, we can filter more easily with `RequireMinClusters` or `RequireSingleCluster`; if the framework was given configs for multi-cluster, we should be able to run multi-cluster tests.

The only suites with this label are under the directory `/tests/integration/multicluster` which we filter with our `go test` command anyway.

tl;dr this is legacy, and we don't need to do it anymore

**Comment by su225** [comment]
Then.. we can clean this up altogether?

**Comment by howardjohn** [comment]
SGTM
